commit 4b50a3d0048d13f6e37126f20f96e8bef262cbe2
Author: Kay Sievers <kay.sievers@vrfy.org>
Date:   Wed Mar 14 01:01:16 2012 +0100

    extras: ata_id - do not log error if HDIO_GET_IDENTITY fails
    
    <tomegun> kay: is this a valid issue: https://bugs.archlinux.org/task/27060 ?
    <kay> tomegun: udev does not really care if that fails
    <tomegun> kay: the suggestion there is to treat EINVAL the same way we treat ENOTTY (i.e. as an info only)
    <tomegun> if it really does not matter it might make sense to avoid bogus bug reports
    <kay> tomegun: done

--- a/extras/ata_id/ata_id.c
+++ b/extras/ata_id/ata_id.c
@@ -532,13 +532,8 @@ int main(int argc, char *argv[])
 	} else {
 		/* If this fails, then try HDIO_GET_IDENTITY */
 		if (ioctl(fd, HDIO_GET_IDENTITY, &id) != 0) {
-			if (errno == ENOTTY) {
-				info(udev, "HDIO_GET_IDENTITY unsupported for '%s'\n", node);
-				rc = 2;
-			} else {
-				err(udev, "HDIO_GET_IDENTITY failed for '%s': %m\n", node);
-				rc = 3;
-			}
+			info(udev, "HDIO_GET_IDENTITY failed for '%s': %m\n", node);
+			rc = 2;
 			goto close;
 		}
 	}
